app/src/main/java/com/infomaniak/mail/utils/FetchMessagesManager.kt

```diff
@@ -154,6 +154,10 @@ class FetchMessagesManager @Inject constructor(
             return true
         }
 
+        notificationMessageUidToLog?.let { notificationMessageUidToLog ->
+            logNotFetchedMessageUid(threadsWithNewMessages, notificationMessageUidToLog, userId, mailbox)
```

Comment:
Also your code doesn't trigger at all when threadsWithNewMessages is empty because you're handling a subsequent firebase notification in a parallel group